server/cluster.go

```diff
@@ -705,104 +706,96 @@ func (c *RaftCluster) putConfig(meta *metapb.Cluster) error {
 	return c.cachedCluster.putMeta(meta)
 }
 
-// GetNamespaces returns all the namespace in etc.d
-func (c *RaftCluster) GetNamespaces() []*Namespace {
-	return c.cachedCluster.getNamespaces()
+// GetNamespaceClassifier returns current namespace classifier.
+func (c *RaftCluster) GetNamespaceClassifier() namespace.Classifier {
+	c.RLock()
```

Comment:
why not saving the classifier in the RaftCluster directly, so we can return it directly, no need Lock here.

Reply:
Addressed, now it is created inside server, seems more reasonable.